src/main/java/com/openlattice/edm/type/PropertyType.java

```diff
@@ -84,8 +84,9 @@ public PropertyType(
         this.multiValued = multiValued.orElse( true );
         this.analyzer = analyzer.orElse( Analyzer.STANDARD );
 
-        if ( EdmPrimitiveTypeKind.Binary == this.datatype && postgresIndexType.isPresent() ) {
-            checkArgument( postgresIndexType.get() == IndexType.NONE, "Indexes are not allowed on Binary datatypes" );
+        if ( EdmPrimitiveTypeKind.Binary == this.datatype ) {
+            this.postgresIndexType = postgresIndexType.orElse( IndexType.NONE );
+            checkArgument( this.postgresIndexType == IndexType.NONE, "Indexes are not allowed on Binary datatypes" );
         }
         this.postgresIndexType = postgresIndexType.orElse( IndexType.BTREE );
```

Comment:
I think this should be in an else otherwise the NONE will be overwritten (or moved above the if statement)

Reply:
Derp